endpoints.tf

```diff
@@ -109,6 +121,18 @@ resource "aws_vpc_endpoint" "s3" {
   tags              = merge(var.tags, { "Name" = "${var.prepend_resource_type ? "endpoint-" : ""}s3-${var.name}" })
 }
 
+# Resources for the S3 VPC interface endpoint
+resource "aws_vpc_endpoint" "s3_interface_endpoint" {
+  count               = var.s3_interface_endpoint != null ? 1 : 0
+  private_dns_enabled = var.s3_interface_endpoint.private_dns_enabled
```

Comment:
set this one explicitly to false (although it is the default for the resource) since it's not supported by s3 interface endpoint.